src/routes/RecipeDetails.js

```diff
@@ -1,12 +1,39 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-const RecipeDetails = props => {
-  const { id } = props.match.params
-  
-  return (
-    <h1 className="text-center">{id}</h1>
-  )
+class RecipeDetails extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      recipe: {},
+    }
+  }
+
+  async componentDidMount() {
+    const response = await fetch(`/api/recipes/${this.props.match.params.id}.json`)
+    const recipe = await response.json()
+    this.setState({ recipe })
+  }
+
+  render() {
+    const { recipe, recipe: {recipeDetail} } = this.state
+
+    return (
+      <div className="text-center">
+        <h2 className="text-secondary text-uppercase mb-4">{recipe.title}</h2>
+        <p>{recipe.description}</p>
+        <img
+          src={recipeDetail && recipeDetail.image}
```

Comment:
This looks very good. 
I would approve this, but I want to introduce you to a new concept called [**optional chaining (`?.`)** ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)
> The **optional chaining** operator (**`?.`**) permits reading the value of a property located deep within a chain of connected objects without having to expressly validate that each reference in the chain is valid. The `?.` operator functions similarly to the `.` chaining operator, except that instead of causing an error if a reference is [nullish](https://developer.mozilla.org/en-US/docs/Glossary/nullish) ([`null`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null) or [`undefined`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined)), the expression short-circuits with a return value of `undefined`. When used with function calls, it returns `undefined` if the given function does not exist.

> This results in shorter and simpler expressions when accessing chained properties when the possibility exists that a reference may be missing. It can also be helpful while exploring the content of an object when there's no known guarantee as to which properties are required.

Or in my own words.
When an object can have `undefined` keys, you can check with `?.` if they exist and call them if they do.
Eg.
```js
const obj = {};
console.log(obj.foo); // prints out `undefined`
console.log(obj.foo.bar); // error `Uncaught TypeError: Cannot read property 'bar' of undefined`
console.log(obj.foo?.bar); // prints out `undefined`
```

In our application we can do
```suggestion
          src={recipeDetail?.image}
```
*Note, alsoe apply it on line 32 😉

Reply:
niiiiiceee